### ISSUE TITLE
[Accessibility]: Should the "You are at the definition" disabled button be part of the Tab hierarchy?

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36566]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33509

### Problem description

When navigating through the Code Intel Popover using Tab, the screen reader skips the "You are the Definition" button which is disabled. This means that pressing Tab after "Precise" directly takes you to the Find References button.

This seems a little confusing from the perspective of a screen reader user, because it's unclear what happened to the Go to Definition button. However, it's possible to navigate to the disabled button using Control + Option + (←|→).

Also, the button is marked clickable (perhaps because of https://github.com/sourcegraph/sourcegraph/issues/35267), but it shouldn't be.

### Expected behavior

I'm not 100% sure, but maybe the disabled button should still be part of the tab hierarchy, so that "You are at the Definition" is announced once you Tab after "Precise".

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
> it's possible to navigate to the disabled button using Control + Option + (←|→).

I think this would be the appropriate way to understand this context. Disabled buttons are not focusable by default (tab behavior should only really be used for interactive elements), IMO making it focusable here wouldn't be preferred.
